test/integration/routes_test.rb

```diff
@@ -0,0 +1,12 @@
+require 'test_helper'
+
+class RoutesTest < ActionDispatch::IntegrationTest
+
+  test "test single-spam-map route" do
```

Comment:
Excellent. Routes are a common point of failure over long periods of time, so I love having these protected.